### DISCUSSION TITLE
[Q&A] Agent unable to report Windows Edition

### DISCUSSION THREAD

**Discussion by dmgpub** [glpi-project/glpi-agent#585] [Q&A] [answered]
### Documentation & bug reporting acknowledgment

Yes, I read it

### Describe your problem

Hi,

I have the latest 1.7 agent installed on a Windows 2019 Server standard edition, the agent is unable to report the OS edition.

Has anyone faced this issue and resolved ?

Here is the os output from inventory task:
```
C:\Program Files\GLPI-Agent>glpi-inventory --partial os
[info] New inventory from xxxx-2024-01-18-09-16-57 for local0
{
   "action": "inventory",
   "content": {
      "bios": {
         "bdate": "2020-11-12",
         "bmanufacturer": "Phoenix Technologies LTD",
         "bversion": "6.00",
         "mmanufacturer": "Intel Corporation",
         "mmodel": "440BX Desktop Reference Platform",
         "smanufacturer": "VMware, Inc.",
         "smodel": "VMware Virtual Platform",
         "ssn": "VMware-xxxx"
      },
      "hardware": {
         "chassis_type": "Other",
         "description": "xxxx",
         "name": "xxx",
         "uuid": "xxx",
         "vmsystem": "VMware",
         "winprodkey": "xxxx"
      },
      "operatingsystem": {
         "arch": "32-bit",
         "dns_domain": "xxxx",
         "fqdn": "xxxx",
         "install_date": "2021-09-09 16:31:41",
         "kernel_name": "MSWin32",
         "name": "Windows",
         "service_pack": "5122",
         "timezone": {
            "name": "Europe/London",
            "offset": "+0000"
         },
         "version": "1809"
      },
      "versionclient": "GLPI-Inventory_v1.7.1"
   },
   "deviceid": "xxxx-2024-01-18-09-16-57",
   "itemtype": "Computer",
   "partial": true
}
[info] Inventory dumped on standard output
```

Many thanks

**Comment by g-bougard** [comment]
Hi @dmgpub 

the agent gets the windows edition from the `Win32_OperatingSystem` WMI object. I know windows server editions handles the system in a different way then desktop editions. Probably this object is not populated the same way.

Anyway, can you share the output of the following command run in an administrative console ?
```
wmic path Win32_OperatingSystem get /format:list
```
Maybe we can use another field to get the windows edition.

**Reply by dmgpub** [reply, marked as answer]
Hi @g-bougard thanks for the quick reply... I was still debugging and discovered that the wbem WMI repository was corrupted on the server, which was causing this issue.

Please excuse the inconvenience.

Many thanks
